wandb/sdk/wandb_settings.py

```diff
@@ -674,6 +679,27 @@ def _path_convert(*args: str) -> str:
         """
         return os.path.expanduser(os.path.join(*args))
 
+    def _get_url_query_string(self) -> str:
+        if self.anonymous != "true":
+            return ""
+        return "?" + urlencode({"apiKey": self.api_key})
+
+    def _get_run_url(self) -> str:
```

Comment:
if you add this logic here... i think it would be great to remove it from the run class and just use the setting in the run class

Reply:
ah yeah, nice! let me check and see if the logic in the run class is being used by anything else.